src/main/java/com/careerfair/q/workflow/database/implementation/StudentFirebaseWorkflowImpl.java

```diff
@@ -84,55 +84,36 @@ public void registerEmployeeToStudent(String studentId, String employeeId)
         Firestore firestore = FirestoreClient.getFirestore();
         Student student = getStudentWithId(studentId);
 
-        if (student.getEmployees() == null) {
-            student.setEmployees(Lists.newArrayList());
+        if (student.employees == null) {
+            student.employees = Lists.newArrayList();
         }
 
-        student.getEmployees().add(employeeId);
+        student.employees.add(employeeId);
         firestore.collection(STUDENT_COLLECTION).document(studentId).update("employees",
-                student.getEmployees());
+                student.employees);
     }
 
     @Override
     public Student updateStudent(String studentId, Student updatedStudent)
             throws FirebaseException {
         Firestore firestore = FirestoreClient.getFirestore();
-        Student firebaseStudent = getStudentWithId(studentId);
-        updateResponseStudent(studentId, firebaseStudent, updatedStudent);
         firestore.collection(STUDENT_COLLECTION).document(studentId).set(updatedStudent);
         return updatedStudent;
     }
 
-    private void updateResponseStudent(String studentId, Student firebaseStudent,
-                                       Student updatedStudent) {
-        if (updatedStudent.getFirstName() == null) {
-            updatedStudent.setFirstName(firebaseStudent.getFirstName());
-        }
-        if (updatedStudent.getLastName() == null) {
-            updatedStudent.setLastName(firebaseStudent.getLastName());
-        }
-        if (updatedStudent.getMajor() == null) {
-            updatedStudent.setMajor(firebaseStudent.getMajor());
-        }
-        if (updatedStudent.getRole() == null) {
-            updatedStudent.setRole(firebaseStudent.getRole());
-        }
-        if (updatedStudent.getBio() == null) {
-            updatedStudent.setMajor(firebaseStudent.getMajor());
-        }
-        if (updatedStudent.getGpa() == null) {
-            updatedStudent.setGpa(firebaseStudent.getGpa());
-        }
-        if (updatedStudent.getGraduationDate() == null) {
-            updatedStudent.setGraduationDate(firebaseStudent.getGraduationDate());
-        }
-        if (updatedStudent.getInternational() == null) {
-            updatedStudent.setInternational(firebaseStudent.getInternational());
-        }
-        updatedStudent.setStudentId(studentId);
-        updatedStudent.setUniversityId(firebaseStudent.getUniversityId());
+    @Override
+    public Student addStudent(Student newStudent) {
+        Firestore firestore = FirestoreClient.getFirestore();
+        CollectionReference studentCollection = firestore.collection(STUDENT_COLLECTION);
+        studentCollection.add(newStudent);
+        return newStudent;
```

Comment:
There is no point in returning the parameter. You should make a call to the DB to get the newly added student and return that